src/MainWindow.vala

```diff
@@ -776,6 +775,10 @@ namespace Terminal {
             }
 
             current_terminal.grab_focus ();
+            if (!closing_current_terminal) { // Otherwise current_terminal will change triggering a name check
```

Comment:
```suggestion
            if (!t != current_terminal) { // Otherwise current_terminal will change triggering a name check
```

Is the `closing_current_terminal` variable necessary?

Reply:
I thought `current_terminal` may have already changed by that point so this seemed safer. I guess the check is only needed when the current_terminal is not closed in which it will not have changed so your way may work - I'll try it.

Reply:
Yes, that seems to work OK - thanks.